### PR TITLE
Add status callback functions after each snapshot generation

### PR DESCRIPTION
**Pull request by YuzhouGuo**
Associate task
https://cyfitech.atlassian.net/browse/CYCL-41

Help to solve the following issue by notifying after each snapshot generation job has been done.
- https://github.com/cyfitech/cris-core/issues/154

### CODE REVIEW

File: src/msg_recorder/recorder.h [hunk 1]
```diff
@@ -46,6 +47,8 @@ class MessageRecorder : public CRNamedNode<MessageRecorder> {
     // Mapping from interval names to snapshot lists. Snapshots are ordered from old to new in the lists.
     std::map<std::string, std::vector<std::filesystem::path>> GetSnapshotPaths();
 
+    void SetPostSnapshotJobFinishCallback(std::function<void()>&& callback);
```
Comment (chenhao94) [inline src/msg_recorder/recorder.h:50]: callback怎么会没有参数呢？你总需要区分finish的是哪个snapshot吧

Comment (chenhao94) [reply src/msg_recorder/recorder.h:50]: SnapshotPostFinish

Post finish 放一起比较容易一次性看明白callback的触发时间

File: src/msg_recorder/recorder.h [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include "cris/core/msg_recorder/record_file.h"
 #include "cris/core/msg_recorder/recorder_config.h"
 
+#include <atomic>
```
Comment (chenhao94) [inline src/msg_recorder/recorder.h:8]: why

**Review by chenhao94 — COMMENTED** [review]
pre-start?

**Comment by chenhao94** [comment]
Please revert because there are many unresolved comments. @YuzhouGuo

File: src/msg_recorder/recorder.h (hunk 2 above)
Comment (YuzhouGuo) [reply src/msg_recorder/recorder.h:8]: 之前漏写了，原来头文件里面也有atomic

**Pull request merged**